app/share/views.py

```diff
@@ -88,6 +96,332 @@ def resources():
 @share.route('/files/<model_run_uuid>')
 @login_required
 def files(model_run_uuid):
+
+    if not os.path.exists(app.config['UPLOAD_FOLDER']):
+        os.makedirs(app.config['UPLOAD_FOLDER'])
+
     "View of file submission for as yet unselected resource to add to"
     model_run_uuid = model_run_uuid
     return render_template('share/files.html', model_run_uuid=model_run_uuid)
+
```

Comment:
Ok, Thanks! Following is what I did so far: When the user selects 'Add files to this resource' option, he/she will be redirected to a page where the user will be asked to upload input file, select model (iSNOBAL, PRMS, reference), input the name of watershed, description of watershed, and select state (NV, NM, Idaho).